docs/source/array-creation.rst

```diff
@@ -130,6 +130,31 @@ As an example, consider loading a stack of images using ``skimage.io.imread``:
 
 See :doc:`documentation on using dask.delayed with collections<delayed-collections>`.
 
+Often it is substantially faster to use ``da.map_blocks`` rather than ``da.stack``
+
+.. code-block:: python
+
+    import glob
+    import skimage.io
+    import numpy as np
+    import dask.array as da
+
+    filenames = sorted(glob.glob('*.jpg'))
+
+    def read_one_image(block_id, filenames=filenames, axis=0):
```

Comment:
Should this be `block_info` instead of `block_id`? See [map_blocks](https://docs.dask.org/en/latest/array-api.html#dask.array.map_blocks)
```
loc = block_info[None]['array-location'][0]
```

Reply:
I guess `block_id` is under-documented. https://github.com/dask/dask/blob/8214861c136199725ae8d35fcab3e3d3e2d63336/dask/array/core.py#L723:L738, but I think it works fine.

Reply:
I think this is equivalent.

```python
import glob
import skimage.io
import numpy as np
import dask.array as da

filenames = sorted(glob.glob('*.tif'))

def read_one_image(block_info=None, filenames=filenames, axis=0):
    # a function that reads in one chunk of data
    path = filenames[block_info[None]["chunk-location"][axis]]
    image = skimage.io.imread(path)
    return np.expand_dims(image, axis=axis)

sample = skimage.io.imread(filenames[0]) # load the first image (assume rest are same shape/dtype)

stack = da.map_blocks(
    read_one_image,
    dtype=sample.dtype,
    chunks=((1,) * len(filenames),  *sample.shape)
)
```

Reply:
ah thanks. It should be in the docstring!

Reply:
Yeah agree. I'll add it in this PR.